### PR TITLE
Sync with the latest wasi-proposal-template

### PR DESCRIPTION
**Pull request by yamt**
https://github.com/WebAssembly/wasi-proposal-template
commit 6601e1064e381370f4f66d1db2d0ccaa006220b6

### CODE REVIEW

**Comment by yamt** [comment]
an alternative to https://github.com/WebAssembly/wasi-threads/pull/23

File: proposal-template.wit.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ More rigorous specification details for the implementer go here, if needed.
 /// Short description
 ///
 /// Explanation for developers using the API.
-api-function-one: function() -> api-type-one
```
Comment (sbc100) [inline proposal-template.wit.md:29]: Why do these templates live in the proposal itself?   Shouldn't it be instances of the templates that live here?  (apologies I've not be following the proposal template stuff).  

i.e. shouldn't the templates get renamed when a proposal is started?

Comment (yamt) [reply proposal-template.wit.md:29]: i guess we should rename them when we start using them.
this PR is meant to be the minimum ci fix.

i personally prefer to remove them until we actually start using them: https://github.com/WebAssembly/wasi-threads/pull/23

Comment (sbc100) [reply proposal-template.wit.md:29]: No, I was just curious.  This pr seems fine.

**Review by loganek — APPROVED** [review]
That looks fine to me as a quick fix; I guess we could now also provide the actual definitions?

**Review by abrown — APPROVED** [review]
I'm fine with this too but I think what I meant by my original comment (wherever I made it) was that we should just write out the WIT for wasi-threads since it is quite simple. This makes CI green, though, so one of us can do that in a future PR... Thanks for fixing this!

**Pull request merged**